### PR TITLE
Get posts for a channel has_next attribute fix

### PR DESCRIPTION
**Pull request by Aryakoste**
#### Summary

Fixed the issue where has_next attribute always returned false. Made has_next attribute pointer and made changes accordingly.  

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/26862
Jira https://mattermost.atlassian.net/browse/MM-57937

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/store/storetest/post_store.go [hunk 1]
```diff
@@ -776,7 +776,7 @@ func testPostStoreGetForThread(t *testing.T, rctx request.CTX, ss store.Store) {
 		require.NoError(t, err)
 		require.Len(t, r1.Order, 3) // including the root post
 		require.Len(t, r1.Posts, 3)
-		assert.True(t, r1.HasNext)
+		require.Nil(t, r1.HasNext)
```
Comment (agnivade) [inline server/channels/store/storetest/post_store.go:779]: These changes seem wrong. The assertion should still be true here. I would assume something like: `assert.True(t, *r1.HasNext)` after a `require.NotNil(t, r1.HasNext)`

**Review by agnivade — CHANGES_REQUESTED** [review]
Also, as I mentioned here: https://github.com/mattermost/mattermost/issues/26862#issuecomment-2075066486, please add a test in the getPostsForChannel API endpoint (see api4/post_test.go) and confirm that the pointer is nil in the response.

**Comment by agnivade** [comment]
@Aryakoste - please check the CI for failures

```
PASS channels/store/localcachelayer.TestPostStore/LocalCache+PostgreSQL/GetForThread/Post_threadmembership_does_not_exist (0.01s)
=== RUN   TestPostStore/LocalCache+PostgreSQL/GetForThread/Pagination
    post_store.go:796: 
        	Error Trace:	/mattermost/server/channels/store/storetest/post_store.go:796
        	Error:      	Should be true
        	Test:       	TestPostStore/LocalCache+PostgreSQL/GetForThread/Pagination
    post_store.go:812: 
        	Error Trace:	/mattermost/server/channels/store/storetest/post_store.go:812
        	Error:      	Should be true
        	Test:       	TestPostStore/LocalCache+PostgreSQL/GetForThread/Pagination
    post_store.go:864: 
        	Error Trace:	/mattermost/server/channels/store/storetest/post_store.go:864
        	Error:      	Should be true
        	Test:       	TestPostStore/LocalCache+PostgreSQL/GetForThread/Pagination
    post_store.go:881: 
        	Error Trace:	/mattermost/server/channels/store/storetest/post_store.go:881
        	Error:      	Should be true
        	Test:       	TestPostStore/LocalCache+PostgreSQL/GetForThread/Pagination
--- FAIL: TestPostStore/LocalCache+PostgreSQL/GetForThread/Pagination (0.02s)
```

The code needs to be changed as well for the tests to work.

**Comment by agnivade** [comment]
@Aryakoste - Please feel free to request a review when this is ready. Thanks!

**Comment by agarciamontoro** [comment]
Hey, @Aryakoste, let us know if we can help with this PR!

**Comment by agarciamontoro** [comment]
Hey @Aryakoste! The first thing we need to check is whether the failing tests are valid. Let me take a look and I'll get back to you!

**Comment by agarciamontoro** [comment]
I didn't have time to check this this week, and I'll be out next week, so it'll have to wait until the week after. Sorry!

File: server/channels/store/storetest/post_store.go [hunk 2]
```diff
@@ -793,7 +793,7 @@ func testPostStoreGetForThread(t *testing.T, rctx request.CTX, ss store.Store) {
 		require.Len(t, r1.Order, 3) // including the root post
 		require.Len(t, r1.Posts, 3)
 		assert.GreaterOrEqual(t, r1.Posts[r1.Order[len(r1.Order)-1]].CreateAt, lastPostCreateAt)
-		assert.False(t, r1.HasNext)
+		assert.True(t, *r1.HasNext)
```
Comment (agarciamontoro) [inline server/channels/store/storetest/post_store.go:796]:
```suggestion
		assert.False(t, *r1.HasNext)
```

File: server/channels/store/storetest/post_store.go [hunk 3]
```diff
@@ -809,7 +809,7 @@ func testPostStoreGetForThread(t *testing.T, rctx request.CTX, ss store.Store) {
 		require.Len(t, r1.Order, 3) // including the root post
 		require.Len(t, r1.Posts, 3)
 		assert.LessOrEqual(t, r1.Posts[r1.Order[1]].CreateAt, firstPostCreateAt)
-		assert.False(t, r1.HasNext)
+		assert.True(t, *r1.HasNext)
```
Comment (agarciamontoro) [inline server/channels/store/storetest/post_store.go:812]:
```suggestion
		assert.False(t, *r1.HasNext)
```

File: server/channels/store/storetest/post_store.go [hunk 4]
```diff
@@ -861,7 +861,7 @@ func testPostStoreGetForThread(t *testing.T, rctx request.CTX, ss store.Store) {
 		require.Len(t, r1.Order, 4) // including the root post
 		require.Len(t, r1.Posts, 4)
 		assert.GreaterOrEqual(t, r1.Posts[r1.Order[len(r1.Order)-1]].CreateAt, lastPostCreateAt)
-		assert.False(t, r1.HasNext)
+		assert.True(t, *r1.HasNext)
```
Comment (agarciamontoro) [inline server/channels/store/storetest/post_store.go:864]:
```suggestion
		assert.False(t, *r1.HasNext)
```

File: server/channels/store/storetest/post_store.go [hunk 5]
```diff
@@ -878,7 +878,7 @@ func testPostStoreGetForThread(t *testing.T, rctx request.CTX, ss store.Store) {
 		require.Len(t, r1.Order, 2) // including the root post
 		require.Len(t, r1.Posts, 2)
 		assert.LessOrEqual(t, r1.Posts[r1.Order[1]].CreateAt, firstPostCreateAt)
-		assert.False(t, r1.HasNext)
+		assert.True(t, *r1.HasNext)
```
Comment (agarciamontoro) [inline server/channels/store/storetest/post_store.go:881]:
```suggestion
		assert.False(t, *r1.HasNext)
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Sorry for the late response!

I just tested this locally, and I believe you need to revert the `False`->`True` changes you made to the tests (I added individual comments below where you need to revert). The only modification the tests need is the dereference of the pointer, which you already did. Can you fix the `False`->`True` changes and push so that we can check whether CI passes? Thanks!

**Comment by Aryakoste** [comment]
Hello @agarciamontoro I have made the changes.

**Review by agarciamontoro — APPROVED** [review]
Things look good now! Thank you, @Aryakoste

**Review by agnivade — APPROVED** [review]
Thanks!

**Comment by agnivade** [comment]
E2E tests passed. Merging.

**Comment by agarciamontoro** [comment]
Thank you for your work (and your patience!), @Aryakoste :tada:

**Pull request merged**